acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,444 @@
+import static org.jenkinsci.test.acceptance.po.Build.Result.SUCCESS;
+
+@WithPlugins({"mailer", "atlassian-bitbucket-server-integration"})
+public class SmokeTest extends AbstractJUnitTest {
+
+    @Rule
+    public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private String bbsAdminCredsId;
+    private BitbucketRepository forkRepo;
+    private String serverId;
+
+    @Before
+    public void setUp() {
+        // BB Access Token
+        PersonalToken bbsToken = createPersonalToken(PROJECT_READ_PERMISSION, REPO_ADMIN_PERMISSION);
+        CredentialsPage credentials = new CredentialsPage(jenkins, DEFAULT_DOMAIN);
+        credentials.open();
+        BitbucketTokenCredentials bbsTokenCreds = credentials.add(BitbucketTokenCredentials.class);
+        bbsTokenCreds.token.set(bbsToken.getSecret());
+        bbsTokenCreds.description.sendKeys(bbsToken.getId());
+        credentials.create();
+
+        // BB Admin user/password
+        credentials.open();
+        UserPwdCredential bbAdminCreds = credentials.add(UserPwdCredential.class);
+        bbsAdminCredsId = "admin-" + randomUUID();
+        bbAdminCreds.setId(bbsAdminCredsId);
+        bbAdminCreds.username.set(BITBUCKET_ADMIN_USERNAME);
+        bbAdminCreds.password.set(BITBUCKET_ADMIN_PASSWORD);
+        credentials.create();
+
+        // Create BB Server entry
+        JenkinsConfig jenkinsConfig = jenkins.getConfigPage();
+        jenkinsConfig.configure();
+        serverId = "bbs-" + randomUUID();
+        new BitbucketPluginConfiguration(jenkinsConfig)
+                .addBitbucketServer(serverId, BITBUCKET_BASE_URL, bbsToken.getId(), bbsAdminCredsId);
+        jenkinsConfig.save();
+
+        // Fork repo
+        forkRepo = forkRepository(PROJECT_KEY, REPO_SLUG, "fork-" + randomUUID());
+    }
+
+    @Test
+    public void testFreeStyleJob() throws IOException, GitAPIException {
+        FreeStyleJob freeStyleJob = jenkins.jobs.create();
+        BitbucketScm bitbucketScm = freeStyleJob.useScm(BitbucketScm.class);
+        bitbucketScm
+                .credentialsId(bbsAdminCredsId)
+                .serverId(serverId)
+                .projectName(forkRepo.getProject().getKey())
+                .repositoryName(forkRepo.getSlug())
+                .anyBranch();
+        freeStyleJob.addTrigger(BitbucketWebhookTrigger.class);
+        freeStyleJob.save();
+
+        // Clone (fork) repo and push new file
+        CredentialsProvider cr =
+                new UsernamePasswordCredentialsProvider(BITBUCKET_ADMIN_USERNAME, BITBUCKET_ADMIN_PASSWORD);
+        File checkoutDir = tempFolder.newFolder("repositoryCheckout");
+        Git gitRepo = cloneGitRepo(cr, checkoutDir, forkRepo);
+
+        final String branchName = "smoke/test";
+        gitRepo.branchCreate().setName(branchName).call();
+        gitRepo.checkout().setName(branchName).call();
+        File jenkinsFile = new File(checkoutDir, "Jenkinsfile");
```

Comment:
/nit - Have a constant Java file to store these reusable name like `Jenkinsfile`